document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoExpressionParser.java

```diff
@@ -0,0 +1,10 @@
+package org.hypertrace.core.documentstore.mongo.parser;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import org.hypertrace.core.documentstore.query.Query;
+
+@AllArgsConstructor(access = AccessLevel.PROTECTED)
+public class MongoExpressionParser {
+  protected final Query query;
```

Comment:
I have observed this it is not used. So, can we remove this as discussed (can be done in follow up)